ui/radar/radar.js

```diff
@@ -0,0 +1,167 @@
+        matchOrNot &= (Number(r[2]) >= (Number(monster['hp']) || 0));
+        if (matchOrNot) {
+          // console.log(r);
+          let mob_name = r[1];
+          let r2 = r[1].match(/(.*?)\((.*?)\)/);
+          if (r2)
+            mob_name = r2[1];
+          let m = {
+            'id': name + '-' + mob_name,
+            'name': r[1],
+            'hp': r[2],
+            'cur_hp': 100,
+            'pos_x': r[3],
+            'pos_y': r[4],
+            'pos': new Point2D(r[3], -r[4]),
+            'pos_z': r[5],
+          };
+          if (targetMonsters.map((x) => x['name']).indexOf(r[1]) == -1) {
+            targetMonsters.push(m);
+          } else {
+            // TODO: update monster position // ?
+          }
+        }
+      }
+    }
+    // update current hp of mob
+    r = e.detail.logs[i].match(/0D:(.*) HP at (\d+)%/);
+    if (r) {
+      for (let i = 0; i < targetMonsters.length; i++) {
+        let monster = targetMonsters[i];
+        if (r[1].match(monster['name']))
+          monster['cur_hp'] = Number(r[2]);
+      }
+    }
+    // removing new combatant
+    r = e.detail.logs[i].match(/04:........:Removing combatant (.*)\. (?:.*) Max HP: (\d+)\./);
+    if (r) {
+      for (let i = 0; i < targetMonsters.length; i++) {
+        let monster = targetMonsters[i];
+        let arrow_id = 'arrow-' + monster['id'];
+        if (monster['dom'] && r[1].match(monster['name'])) {
+          if (monster['cur_hp'] > 50)
+            continue;
+          monster['dom'].remove();
+          targetMonsters.splice(i, 1);
+          break;
+        }
+      }
+    }
+    // debug:
+    r = e.detail.logs[i].match(/:record pos/);
+    if (r) {
+      let Vt = new Point2D(playerPos['x'], playerPos['y']);
+      target.push(Vt);
+      console.log(playerPos);
+      console.log(target);
+    }
+    r = e.detail.logs[i].match(/:reset pos/);
+    if (r) {
+      target = [];
+      console.log(playerPos);
+      console.log(target);
+    }
+  }
+});
+
+document.addEventListener('onPlayerChangedEvent', function(e) {
+  playerPos['x'] = e.detail.pos.x;
+  playerPos['y'] = -e.detail.pos.y;
+  let tr;
+  for (let i = 0; i < targetMonsters.length; i++) {
+    let monster = targetMonsters[i];
+    let arrow_id = 'arrow-' + monster['id'];
+    if (monster['dom'] == null) {
+      // add dom
+      let tr = document.createElement('tr');
+      let th = document.createElement('th');
+      let img = document.createElement('img');
+      img.setAttribute('id', arrow_id);
+      img.setAttribute('src', 'arrow.png');
+      img.setAttribute('class', 'radar-image-40');
+      th.appendChild(img);
+      th.setAttribute('style', 'max-width: 100px');
+      tr.appendChild(th);
+      th = document.createElement('th');
+      let text = document.createElement('div');
+      text.innerHTML = monster['name'];
+      th.appendChild(text);
+      tr.appendChild(th);
+      $('#radar-table').prepend(tr);
```

Comment:
I think I would rather not have the jquery dependency here.  (I know it's used in one dps file, but I'd like to keep it there or remove it.)  That would solve the eslint problem, although maybe not in the way you hoped.

It looks like you're only using it for this `prepend`, and the `rotate` below? You can use the built in `insertBefore` to do the equivalent of `prepend`.  You should also be able to add a css style for `transform: rotate` to do the rotation, I think.

Reply:
Ok I removed all the jquery libs.